contentacms/contenta_jsonapi
# Create temporary key pair and Simple OAuth client

Task: #56


Creates a temporary RSA key pair and demo OAuth2 client.

### Notes

Not quite working yet.

It seems that the Simple OAuth module is not yet installed when `contenta_jsonapi_install()` is called. Therefore, the configuration created in the `_install()` is not able to be saved.

In Reservoir, the modules are installed via the profile's .info.yml file's `dependencies`. I'm guessing this is why the code this PR is based on works there. What is the reasoning behind doing the config install in Contenta? I've never encountered that approach before.

Is there a workaround for this?

### Test Instructions

1. Install Contenta
1. Browse to `/admin/config/people/simple_oauth` and verify keys and timeouts are set
1. Browse to `/admin/config/people/simple_oauth/oauth2_client` and verify "demo app" client exists

### QA
